docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,596 @@
+      Object.assign(error, {
+        code: AUTHENTICATION_STRATEGY_NOT_FOUND,
+      });
+      throw error;
+    }
+    return strategy;
+  }
+
+  async findAuthenticationStrategy(name: string) {
+    const strategies = await this.authenticationStrategies();
+    const matchingAuthStrategy = strategies.find(a => a.name === name);
+    return matchingAuthStrategy;
+  }
+}
+```
+
+{% include note.html content="
+The specific <b>name</b> that it looks for is a unique name you specify as a parameter in an <b>authentication decorator</b> applied to a controller endpoint. This decorator is discussed in the next section.
+" %}
+
+In order for your custom authentication strategy to be **found**, it needs to be
+**registered**.
+
+**Registering** a custom authentication strategy `BasicAuthenticationStrategy`
+as an extension of the
+`AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME` extension
+point in an application `application.ts` is as simple as:
+
+```ts
+registerAuthenticationStrategy(this, BasicAuthenticationStrategy);
+```
+
+## Authentication Decorator
+
+Once you have a **registered** authentication strategy with a **unique** name,
+then you are ready to use the
+[Authentication Decorator](https://loopback.io/doc/en/lb4/Decorators_authenticate.html)
+on your controller endpoints.
+
+The decorator's syntax is :
+
+```ts
+@authenticate(strategyName: string, options?: Object)
+```
+
+When the options object is specified, it must be relevant to that particular
+strategy.
+
+To **secure** a specific endpoint, you simply need to add the decorator and
+specify the `name` of a registered authentication strategy.
+
+Here is an example of the decorator using the 'basic' authentication strategy we
+defined earlier, **without** options, for the endpoint `/whoami` in a controller
+named `WhoAmIController`.
+
+```ts
+import {inject} from '@loopback/context';
+import {
+  AuthenticationBindings,
+  UserProfile,
+  authenticate,
+} from '@loopback/authentication';
+import {get} from '@loopback/rest';
+
+export class WhoAmIController {
+  constructor(
+    @inject(AuthenticationBindings.CURRENT_USER)
+    private userProfile: UserProfile,
+  ) {}
+
+  @authenticate('basic')
+  @get('/whoami')
+  whoAmI(): string {
+    return this.userProfile.id;
+  }
+}
+```
+
+An example of the decorator when options **are** specified looks like this:
+
+```ts
+@authenticate('basic', { propertyOne : "foo", propertyTwo:"bar"})
+```
+
+{% include tip.html content="
+To avoid repeating the same options in the <b>@authenticate</b> decorator for many endpoints in a controller, you can instead define global options which can be injected into an authentication strategy thereby allowing you to avoid specifying the options object in the decorator itself. For controller endpoints that need to override a global option, you can specify it in an options object passed into the decorator. Your authentication strategy would need to handle the option overrides.
```

Comment:
It would be good if we can show a code example here. While I also realized we don't have an auth option sample yet, maybe use the greeter extension https://github.com/strongloop/loopback-next/tree/master/examples/greeter-extension#implement-an-extension ?

Reply:
@jannyHou  ^^ ,

Can we point to a `unit test` that passes in options? https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/__tests__/unit/providers/auth-metadata.provider.unit.ts#L16

Reply:
@emonddr Cool the unit test is much more relevant. Looks good to me.

Reply:
@jannyHou , I've changed my mind about pointing to the unit test. It doesn't really show global options being overridden by `request` options from the decorator. Perhaps after this PR lands, we can have a follow up PR to add a section about injecting global options into the  strategy, and then the strategy checking for overriding options from the decorator and then overriding the global options. I have an example of this I can dig up from one of my workspaces. But I won't be able to prepare it nicely today for @raymondfeng  to review.

Reply:
@emonddr Sure it would be good to have an example and point to it. +1 for improving it in another PR.